crates/services/p2p/src/request_response/messages.rs

```diff
@@ -38,6 +38,12 @@ pub enum ResponseMessageErrorCode {
     /// The peer sent an empty response using protocol `/fuel/req_res/0.0.1`
     #[error("Empty response sent by peer using legacy protocol /fuel/req_res/0.0.1")]
     ProtocolV1EmptyResponse = 0,
+    #[error("The requested range is too large")]
+    RequestedRangeTooLarge = 1,
+    #[error("Timeout while processing request")]
+    Timeout = 2,
+    #[error("Sync processor is out of capacity")]
+    SyncProcessorOutOfCapacity = 3,
```

Comment:
Also, maybe we need to add `Unknown(u32)` for backward compatibility in the case if we can't deserialize the error.

Reply:
Tried something in 5ea9b30e4d409b85f76aa40cc60c52d409b11ec9. There is an unknown variant, but the error code is erased away. The reason for this is that deserializing unknown values in serde has been recently implemented via the attribute `serde(untagged)`, but this is not supported by postcard (fails with error `WontImplement`).  See 5ea9b30e4d409b85f76aa40cc60c52d409b11ec9 for the discussion on `serde(untagged)`. There are other alternatives proposed, such as nested enums or custom deserialization functions for a variant, but they won't either work with postcard or are too complicated imho. 

Adding an `Unknown` variant which abstracts away the error code does at least ensure backwards compatibility.